### DISCUSSION TITLE
I keep getting To build this project, the following workloads must be installed: aspire

### DISCUSSION THREAD

**Discussion by mohamadlawand087** [microsoft/aspire#3879] [Q&A]
I have an application which is running perfectly on Aspire Preview 4. After I upgraded to Aspire Preview 5 I keep getting the following error 

<img width="1948" alt="Screenshot 2024-04-23 at 06 29 01" src="https://github.com/dotnet/aspire/assets/76842090/799a9f62-dac9-483a-8780-98a6e898db77">

This is my dotnet workload list
<img width="799" alt="Screenshot 2024-04-23 at 06 29 32" src="https://github.com/dotnet/aspire/assets/76842090/3ed00005-e3dc-414a-bbf7-acd054fc1cb7">

I have tried create a starter project and I still get the same error

I am using an M1 Mac

**Comment by davidfowl** [comment]
The trickiest thing about understanding workloads is that they are installed per SDK. Do you have a single version of the dotnet sdk installed on your machine?

**Reply by mohamadlawand087** [reply]
I have .Net 6 and .net 8

**Reply by mohamadlawand087** [reply]
As well preview 9

**Reply by davidfowl** [reply]
In the failing environment, it's likely the workload isn't installed.

**Reply by mohamadlawand087** [reply]
I just uninstalled 6 and 9 and it's still failing

**Reply by davidfowl** [reply]
Run dotnet --info and dump the output? What environment is it failing in?

**Reply by mohamadlawand087** [reply]
`.NET SDK:
 Version:           8.0.104
 Commit:            034f91fcc0
 Workload version:  8.0.100-manifests.2f72bdd4

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  14.4
 OS Platform: Darwin
 RID:         osx-arm64
 Base Path:   /opt/homebrew/Cellar/dotnet/8.0.4/libexec/sdk/8.0.104/

.NET workloads installed:
 Workload version: 8.0.100-manifests.2f72bdd4
 [aspire]
   Installation Source: SDK 8.0.100
   Manifest Version:    8.0.0-preview.5.24201.12/8.0.100
   Manifest Path:       /Users/mohamad/.dotnet/sdk-manifests/8.0.100/microsoft.net.sdk.aspire/8.0.0-preview.5.24201.12/WorkloadManifest.json
   Install Type:        FileBased


Host:
  Version:      8.0.4
  Architecture: arm64
  Commit:       2d7eea2529

.NET SDKs installed:
  8.0.104 [/opt/homebrew/Cellar/dotnet/8.0.4/libexec/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 8.0.4 [/opt/homebrew/Cellar/dotnet/8.0.4/libexec/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 8.0.4 [/opt/homebrew/Cellar/dotnet/8.0.4/libexec/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  DOTNET_ROOT       [/opt/homebrew/Cellar/dotnet/8.0.4/libexec]
`

**Reply by mohamadlawand087** [reply]
its failing in the my local dev environment

**Reply by davidfowl** [reply]
Does the command line work? Can you use dotnet run on the apphost project?

**Reply by mohamadlawand087** [reply]
I made it work from the command line after restarting and deleting all bin / obj folders 

but it still doesn't work from the IDE

**Reply by davidfowl** [reply]
What IDE?

**Reply by mohamadlawand087** [reply]
Rider and also from VS Code
